javalin/src/main/java/io/javalin/http/servlet/JavalinServletContext.kt

```diff
@@ -141,7 +144,12 @@ class JavalinServletContext(
         userFutureSupplier = future
     }
 
-    override fun skipRemainingHandlers(): Context = also { tasks.clear() }
+    override fun skipRemainingHandlers(): Context = also {
+        tasks.clear()
+        if (handlerType() == WEBSOCKET_BEFORE_UPGRADE) {
+            throw UnauthorizedResponse()
```

Comment:
Maybe this should be something more specific, like "EndLifecycleException" ?

Reply:
I picked something that would not cause the unexpected exception error message from Javalin. So it should be something that extends HttpResponseException for simplicity.

Reply:
But you're catching it anyway? 🤔

Reply:
The router does that type of logging when it handles an exception without an exception mapper. Either we create an extra method which does not do that or we use an exception that "follows the rules." I just want to avoid spamming the logs of users with an internal Javalin exception.

Reply:
You are catching it here:

```kotlin
try {
    beforeUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
    super.service(req, res) // everything is okay, perform websocket upgrade
    afterUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
} catch (e: Exception) {
    cfg.pvt.internalRouter.handleHttpException(upgradeContext, e)
}
```

When the user calls `skipRemainingHandlers`, they are not saying "this is unauthorized". If they want to say that, they will throw the exception themselves. I think it should be:

```kotlin
try {
    beforeUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
    super.service(req, res) // everything is okay, perform websocket upgrade
    afterUpgradeHandlers.forEach { it.handle(upgradeContext, requestUri) }
} catch (e: Exception) {
    if endlifecycleexception return // don't do anything if user ended lifecycle
    cfg.pvt.internalRouter.handleHttpException(upgradeContext, e)
}

Reply:
I have switched this over, but the change currently leads to a status code of 200 without the creation of a websocket connection server side. That feels not correct to me from an users perspective.